Week3/week3.txt

```diff
@@ -0,0 +1,143 @@
+Write a python script to perform CRUD operation on following table of "ESM.db" database.
+
+Employee ( eid, ename, dept, basic, branch). eid must be primary key.
+
+Department : Account, Inventory, IT, HR.
+
+Peform Following Operation.
+
+1. Create table
+
+	import sqlite3 as sqlite
+	con=sqlite.connect("c:\\sqlite3\\ems.db")
+	E="create table employee37(eid int primary key,ename text,dept text,basic int,branch text)"
+	cur=con.cursor()
+	cur.execute(E)
+	con.commit()
+	con.close()
+
+2. Insert 5 Records directly, 5 records using tuple and 5 records using taking input from user.
+
+	import sqlite3 as sqlite
+	con=sqlite.connect("c:\\sqlite3\\ems.db")
+	cur=con.cursor()
+	#-->insert record directly
+	q="insert into employee37 values(1,'om','Account',8000,'surat'),
+                                      (2,'sai','Inventory'5500,'vyara'),
+	                              (3,'ram','HR',13000,'baroda'),
+                                      (4,'radha','IT',20000,'valsad'),
+                                      (5,'krishna','HR',75000,'valod');"
+	cur.execute()
+	con.commit()
+	#-->insert record using tuple
+	m=[(6,'preet','Inventory',8500,'valsad'),
+   	     (7,'meet','HR',9500,'vesu'),
+   	     (8,'rohit','IT',5500,'surat'),
+	     (9,'jeet','Account',10000,'bodhan'),
+  	    (10,'riya','HR',7500,'ryan')]
+	b="insert into employee37 values(?,?,?,?,?)"
+	cur.executemany(b,m)
+	con.commit()
+	# insert record through user input
+	c="insert into employee37 values(?,?,?,?,?)"
+	l=[]
+	for a in range(5):
+    		no=int(input("Enter emp id:"))
+    		name=input("Enter emp name:")
+    		dep=input("Enter emp department:")
+    		basic=int(input("Enter emp basic:"))
+   		 branch=input("Enter emp branch")
+    		t=(no,name,dep,basic,branch)
+    		l.append(t)
+	cur.executemany(c,l)
+	con.commit()
+	con.close()
+
+3. Update records who are from "Surat" branch with increment in salary 10%.
+-->	
+	import sqlite3 as sqlite
+	con=sqlite.connect("c:\\sqlite3\\ems.db")
+	u="update employee37 set basic=basic+(basic*10)/100 where branch='surat'"
+	cur=con.cursor()
+	cur.execute(u)
+	con.commit()
+	con.close()
+	
+4. Print All records.
+-->
+	import sqlite3 as sqlite
+	con=sqlite.connect("c:\\sqlite3\\ems.db")
+	se="select * from employee37"
+	cur=con.cursor()
+	cur.execute(se)
+	r=cur.fetchall()

```

Comment:
what is the use of cur.fetchall? can you explain?

Reply:
Fetchall() is a method that fetches all the remaining tuples from the last executed statement from a table.